### PR TITLE
Hook up add database references dialog

### PR DESCRIPTION
**Pull request by kisantia**
This PR hooks up the add database reference dialog so that the references get added to the sqlproj and gets rid of the old quick pick flow for adding a database reference.

Support for creating sqlcmd vars and references for different database, different server will be added in a separate PR.

### CODE REVIEW

File: extensions/sql-database-projects/src/models/project.ts [hunk 1]
```diff
@@ -432,32 +436,41 @@ export class Project {
 		throw new Error(constants.unableToFindObject(path, constants.folderObject));
 	}
 
+	private addSystemDatabaseReferenceToProjFile(entry: SystemDatabaseReferenceProjectEntry): void {
+		const systemDbReferenceNode = this.projFileXmlDoc.createElement(constants.ArtifactReference);
+
+		// if it's a system database reference, we'll add an additional node with the SSDT location of the dacpac later
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/models/project.ts:442]: you already added that below right? (later meaning below?)

Comment (kisantia) [reply extensions/sql-database-projects/src/models/project.ts:442]: yup the later here means a few lines below

File: extensions/sql-database-projects/src/models/project.ts [hunk 2]
```diff
@@ -466,17 +479,19 @@ export class Project {
...
+		const falseTextNode = this.projFileXmlDoc.createTextNode('False');
 		suppressMissingDependenciesErrorNode.appendChild(falseTextNode);
 		referenceNode.appendChild(suppressMissingDependenciesErrorNode);
 
-		if (name) {
-			let databaseVariableLiteralValue = this.projFileXmlDoc.createElement(constants.DatabaseVariableLiteralValue);
-			let databaseTextNode = this.projFileXmlDoc.createTextNode(name);
-			databaseVariableLiteralValue.appendChild(databaseTextNode);
-			referenceNode.appendChild(databaseVariableLiteralValue);
+		// TODO: add support for sqlcmd vars and server
```
Comment (Benjin) [inline extensions/sql-database-projects/src/models/project.ts:489]: Can you create a GH issue for this and add the URL here?

File: extensions/sql-database-projects/src/test/project.test.ts [hunk 3]
```diff
@@ -192,21 +192,21 @@ describe('Project: sqlproj content operations', function (): void {
...
-		await project.addSystemDatabaseReference(SystemDatabase.msdb);
+		await project.addSystemDatabaseReference({ databaseName: 'msdb', systemDb: SystemDatabase.msdb });
 		should(project.databaseReferences.length).equal(2, 'There should be two database references after adding a reference to msdb');
 		should(project.databaseReferences[1].databaseName).equal(constants.msdb, 'The database reference should be msdb');
 		// make sure reference to SSDT msdb dacpac was added
 		projFileText = (await fs.readFile(projFilePath)).toString();
 		should(projFileText).containEql(convertSlashesForSqlProj(project.getSystemDacpacSsdtUri(constants.msdb).fsPath.substring(1)));
 
-		await project.addDatabaseReference(Uri.parse('test.dacpac'), DatabaseReferenceLocation.sameDatabase);
+		await project.addDatabaseReference({ dacpacFileLocation: Uri.file('test.dacpac'), databaseLocation: DatabaseReferenceLocation.sameDatabase });
```
Comment (Benjin) [inline extensions/sql-database-projects/src/test/project.test.ts:209]: the master and msdb references confirm that the entry's correct in the sqlproj file; is there a way to do that here as well, but for dacpacs? (or is that covered by another test?)

File: extensions/sql-database-projects/src/models/project.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/models/project.ts]: added url for https://github.com/microsoft/azuredatastudio/issues/12036

File: extensions/sql-database-projects/src/test/project.test.ts (hunk 3 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/test/project.test.ts:211]: those checks above were just meant to verify that an entry was also added for the SSDT location of the system dacpacs, but now I added checks to verify that the ADS system dacpacs and the test.dacpac got added to the sqlproj too

Comment (Benjin) [reply extensions/sql-database-projects/src/test/project.test.ts:211]: Perfect; thanks!

**Pull request merged**